docs/src/design/decoder/constraints.md

```diff
@@ -32,10 +32,17 @@ In the sections below, we describe constraints according to their logical groupi
 
 ## General constraints
 
+Make sure that `extra` register, used for degree reduction during flags computation of very high-degree operations, is set to $1$ when both $b_6$ and $b_5$ columns set to $1$:
+
+> $$
+b_6 \cdot b_5 \cdot (1 - extra) = 0 \text{ | degree} = 3
+$$
```

Comment:
I'm not sure we need to describe this constraint here. I think it belongs to the `op_constraints.md` file (which you also have).

Reply:
It's a little bit confusing for me, because, on the contrary, I thought that since we enforce constraints for the _decoder_ registers, it is necessary to add docs in decoder, but not necessary in stack. Is this because the main place where we use these registers is stack, during the flag calculation?

Reply:
I think about this way: we describe all `op_bit` related logic in the `op_constraints.md` file, this would naturally fit there as well. Basically, while these registers are a part of the decoder, they are significant enough to be in their own section.